### PR TITLE
waynergy: init at 0.0.13

### PR DESCRIPTION
**Pull request by themaxhero**
###### Description of changes
Include waynergy package for https://github.com/r-c-f/waynergy

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bachp — CHANGES_REQUESTED** [review]
The desktop file installed under `share/applications/waynergy.desktop` containes a hardcoded path to `/usr/bin`. This will not work and should be replaced.

One way is like in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/backup/vorta/default.nix#L39 but I'm not sure if there is better way.

**Comment by PedroHLC** [comment]
Result of `nixpkgs-review pr 196478` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waynergy</li>
  </ul>
</details>

**Review by PedroHLC — CHANGES_REQUESTED** [review]
Bachp is right:
```console
[nix-shell:~/.cache/nixpkgs-review/pr-196478/results/waynergy]$ find . -type f | xargs grep -a '/usr'
./share/applications/waynergy.desktop:Exec=/usr/bin/waynerg
```

`substituteInPlace` is generally the solution, in either `postPatch` (when source is available prebuild) or `postInstall`.

File: pkgs/tools/wayland/waynergy/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+, libressl
+}:
+stdenv.mkDerivation rec {
+  pname = "waynergy";
+  version = "0.0.13";
+
+  src = fetchurl {
+    url = "https://github.com/r-c-f/waynergy/archive/refs/tags/v${version}.tar.gz";
+    sha256 = "0nblkmaj1zv9i4ghi1q68fr0bbg0qmdxnpyiqbl6zpc1hwknhwg7";
+  };
```
Comment (PedroHLC) [inline pkgs/tools/wayland/waynergy/default.nix:15]: Also, you can use fetchGitHub here.

**Review by PedroHLC — CHANGES_REQUESTED** [review]

File: pkgs/tools/wayland/waynergy/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, libressl
+}:
+stdenv.mkDerivation rec {
+  pname = "waynergy";
+  version = "0.0.13";
+
+  src = fetchFromGitHub {
+    owner = "r-c-f";
+    repo = "waynergy";
+    rev = "refs/tags/v${version}";
```
Comment (PedroHLC) [inline pkgs/tools/wayland/waynergy/default.nix:15]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/wayland/waynergy/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 = "sha256-eTY7tktUmoTZO3w9uP1P8cIz0mmFiWm5YFGVAS6JwwE=";
```
Comment (PedroHLC) [inline pkgs/tools/wayland/waynergy/default.nix:16]: You either do (preferred):
```suggestion
    hash = "sha256-eTY7tktUmoTZO3w9uP1P8cIz0mmFiWm5YFGVAS6JwwE=";
```
Or use a base16/base32/base64 **unprefixed** sha256 with `sha256 = `.

See more at: https://nixos.wiki/wiki/Nix_Hash#Usage

**Review by PedroHLC — CHANGES_REQUESTED** [review]

**Review by PedroHLC — APPROVED** [review]

**Review by PedroHLC — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

File: pkgs/tools/wayland/waynergy/default.nix (hunk 2 above)
Comment (Artturin) [reply pkgs/tools/wayland/waynergy/default.nix]: refs/tags was correct too fyi

**Pull request merged**